spec/ruby_terraform/commands/apply_spec.rb

```diff
@@ -11,280 +14,36 @@
     RubyTerraform.reset!
   end
 
-  it 'calls the terraform apply command passing the supplied directory' do
-    command = RubyTerraform::Commands::Apply.new(binary: 'terraform')
-
-    expect(Open4).to(
-        receive(:spawn)
-            .with('terraform apply some/path/to/terraform/configuration', any_args))
-
-    command.execute(directory: 'some/path/to/terraform/configuration')
-  end
-
-  it 'calls the terraform apply command passing the supplied plan' do
-    command = RubyTerraform::Commands::Apply.new(binary: 'terraform')
-
-    expect(Open4).to(
-        receive(:spawn)
-            .with('terraform apply some/path/to/terraform/plan', any_args))
-
-    command.execute(plan: 'some/path/to/terraform/plan')
-  end
-
-  it 'prefers the plan if both plan and directory provided' do
-    command = RubyTerraform::Commands::Apply.new(binary: 'terraform')
-
-    expect(Open4).to(
-        receive(:spawn)
-            .with('terraform apply some/path/to/terraform/plan', any_args))
-
-    command
-        .execute(
-            directory: 'some/path/to/terraform/configuration',
-            plan: 'some/path/to/terraform/plan')
-  end
-
-  it 'defaults to the configured binary when none provided' do
-    command = RubyTerraform::Commands::Apply.new
-
-    expect(Open4).to(
-        receive(:spawn)
-            .with('path/to/binary apply some/path/to/terraform/configuration', any_args))
-
-    command.execute(directory: 'some/path/to/terraform/configuration')
-  end
-
-  it 'adds a var option for each supplied var' do
-    command = RubyTerraform::Commands::Apply.new(binary: 'terraform')
-
-    expect(Open4).to(
-        receive(:spawn)
-            .with("terraform apply -var 'first=1' -var 'second=two' some/configuration", any_args))
-
-    command.execute(
-        directory: 'some/configuration',
-        vars: {
-            first: 1,
-            second: 'two'
-        })
-  end
-
-  it 'correctly serialises list/tuple vars' do
-    command = RubyTerraform::Commands::Apply.new(binary: 'terraform')
-
-    expect(Open4).to(
-        receive(:spawn)
-            .with("terraform apply -var 'list=[1,\"two\",3]' some/configuration", any_args))
-
-    command.execute(
-        directory: 'some/configuration',
-        vars: {
-            list: [1, "two", 3]
-        })
-  end
-
-  it 'correctly serialises map/object vars' do
-    command = RubyTerraform::Commands::Apply.new(binary: 'terraform')
-
-    expect(Open4).to(
-        receive(:spawn)
-            .with("terraform apply -var 'map={\"first\":1,\"second\":\"two\"}' some/configuration", any_args))
-
-    command.execute(
-        directory: 'some/configuration',
-        vars: {
-            map: {
-                first: 1,
-                second: "two"
-            }
-        })
-  end
-
-  it 'correctly serialises vars with lists/tuples of maps/objects' do
-    command = RubyTerraform::Commands::Apply.new(binary: 'terraform')
+  terraform_command = 'apply'
```

Comment:
Out of interest, why not use a `let` for this?

Reply:
An unfortunate downside of shared examples - the parameters need to be hardcoded or defined prior to test execution otherwise you end up with:

```
An error occurred while loading ./spec/ruby_terraform/commands/apply_spec.rb.
Failure/Error: it_behaves_like 'a command with an argument', [terraform_command, :directory]
  `terraform_command` is not available on an example group (e.g. a `describe` or `context` block). It is only available from within individual examples (e.g. `it` blocks) or from constructs that run in the scope of an example (e.g. `before`, `let`, etc).
# ./spec/ruby_terraform/commands/apply_spec.rb:18:in `block in <top (required)>'
# ./spec/ruby_terraform/commands/apply_spec.rb:3:in `<top (required)>'

Finished in 0.00003 seconds (files took 0.19267 seconds to load)
0 examples, 0 failures, 1 error occurred outside of examples
```

(output after replacing it with `  let(:terraform_command) { 'apply' }`)

Reply:
Admittedly, I could just copy paste `'apply'` into each of the shared example calls in this spec and remove the local variable - but it made copying and pasting between specs a lot easier with it in a 'common' variable. (so yes a level of developer laziness in this one :-) )

Reply:
No need to inline, was more to satisfy my curiosity :)